### ISSUE TITLE
Where can i download PASCAL VOC 07++12+coco dataset

### ISSUE THREAD

**Issue by David-Jin** [weiliu89/caffe#335]
Hi：
     I want to reproduce the result of SSD512 model in 07++12 and 07+12+COCO and 07++12+COCO
but i can't find these datasets except PASCAL VOC 07+12 dataset.
     Are there any scripts to create the "trainval.txt, test.txt, test_name_size.txt and label_map.txt" files? if these data set (07++12 and 07+12+COCO and 07++12+COCO) is got.
     Anyone can help?

**Comment by youngwanLEE** [comment]
@David-Jin Did you solve this problem ? I want to solve this,too.

**Comment by David-Jin** [comment]
No,I didn't. No one  respond to  this  problem.

???? iPhone

? 2016?12?10??12:02?youngwanLEE <notifications@github.com<mailto:notifications@github.com>> ???


@David-Jin<https://github.com/David-Jin> Did you solve this problem ? I want to solve this,too.

-

**Comment by weiliu89** [comment]
I trained from COCO first and then finetune on 07+12

**Comment by David-Jin** [comment]
@weiliu89   so is there no such datasheet named 07++12  ？

**Comment by weiliu89** [comment]
07++12 includes 07 test in training as well. You can refer to VOC0712 and see how to create the lmdb for 07++12.

**Comment by KevinYuk** [comment]
Hi @weiliu89,

Regarding to "train from COCO and then finetune on VOC0712", I found that channel num of COCO-SSD is different from that of VOC0712-SSD. e.g.: conv8_2_mbox_conf of COCO-SSD is 324(aka: 81 x 4), and that of VOC0712-SSD is 84(aka: 21 x 4). We know that it's because class of COCO and VOC0712 is different.

Could I **directly** train COCO with COCO-SSD and then finetune on VOC0712 with VOC0712-SSD even if their channels are not the same? I think the VOC0712-SSD network will automatically adjust itself for that. Am I right?

Thanks a lot.

**Comment by KevinYuk** [comment]
Hi @weiliu89,

After carefully read the related issues, I found that you said:
1. If there is overlap between your classes and VOC classes, you should extract the corresponding classification kernels from the SSD VOC model. (from #224 )
2. If you try to fine-tune from an existing SSD model (e.g. SSD trained from ILSVRC DET or COCO), then you need to change the name for your confidence prediction layers (to not load the confidence predictions from an existing SSD model), and set a smaller learning rate.  (from #120 )

But there is overlap classes between COCO and VOC(e.g.: class person, bicycle etc. I am trying fine-tune from COCO to VOC).  If follow your advice 1, I should extract corresponding classification kernels from the COCO-SSD model; If follow your advice 2, I can directly change the name of all the confidence prediction layers, which means I drop the previous classification kernels weights even for the overlap classes.

Q1. My understanding is that both advices are OK. For simplicity, we can follow advice 2(But advice 1 here may reach higher mAP). We can let networks learn again for the overlap classes. Right?
Q2. smaller learning rates are for all the confidence prediction layers(aka: mbox_conf sub string layers), right?

Thanks a lot.

**Comment by ujsyehao** [comment]
Hi, @KevinYuk Have you solved the problem?

**Comment by ujsyehao** [comment]
How to train 07+12+coco dataset?

**Comment by KevinYuk** [comment]
@ujsyehao Yes. I solved it. 
If you want to get higher accuracy without changing your SSD network topo, you should:
1. train you SSD based on coco dataset firstly(should change channel num to fit coco object category);
2. Drop extended layers weights(just simply rename layers names and change channel num) of your SSD;
3. Fine tune from step2 based on voc07+12.
Thanks.

**Comment by ujsyehao** [comment]
@KevinYuk Thank you for your reply!
I have some problems listed below:

- I train SSD on coco dataset with the pretrain model(weiliu provides).

- It means that follow your advice 2, rename all confidence prediction layers. Do I need to rename location regression layers?

- When I finetune trained SSD-COCO model on voc0712 dataset, Do I  need to use smaller learning rate for all confidence layers? (That is your second question)

**Comment by KevinYuk** [comment]
I train SSD on coco dataset with the pretrain model(weiliu provides).
A: I train base imagenet model(vgg or resnet) by myself. However, If you just use weiliu provided, I think it's ok.

It means that follow your advice 2, rename all confidence prediction layers. Do I need to remove location regression layers?
A: I don't remove location regression layers, and still reach good acc boost. However, you can try to remove them to do comparison and tell me the result :) .

When I finetune trained SSD-COCO model on voc0712 dataset, Do I need to use smaller learning rate for all confidence layers? (That is your second question)
A: I don't use smaller lr. However, you can try more lr.

**Comment by ujsyehao** [comment]
Thank you! @KevinYuk 
I will do experiments follow your advice, Let's keep in touch!
This is my mail (rawk.vx@gmail.com).

**Comment by ujsyehao** [comment]
I follow your advice 

> remove confidence prediction layers, I use lr = 0.0005.

The result is 3 percentage points lower than author reports. I am trying to adjustment.

**Comment by LiMengyang990726** [comment]
Hi, may I ask if I am able to obtain the config file for SSD512 model?
